packages/utils/src/schema/toPathSchema.ts

```diff
@@ -54,7 +54,7 @@ export default function toPathSchema<
 
   if (
     ADDITIONAL_PROPERTIES_KEY in schema &&
-    schema[ADDITIONAL_PROPERTIES_KEY] === true
+    schema[ADDITIONAL_PROPERTIES_KEY] !== false
```

Comment:
We use the path schema to decide which objects to drop when we use `omitExtraData`. For some reason, a change here uncovered a case where the additionalProperty fields would be dropped by `omitExtraData` + `liveOmit` if the `additionalProperties` schema was an object. This fixes that regression.